File: nodejs/src/robot_controller.ts
import { Client } from "./client";
import { IotDeviceService } from "./iot_device";

export class RobotController extends Client {
  public IotDevice: IotDeviceService;

  constructor(config: any) {
    super(config);
    this.IotDevice = new IotDeviceService(this);
  }
}

File: java/src/main/java/cn/boticz/masrobo/RobotController.java
package cn.boticz.masrobo;

import cn.boticz.masrobo.service.IotDeviceService;
import cn.boticz.masrobo.client.Config;
import cn.boticz.masrobo.client.OpenHttpClient;


public class RobotController {
    private final IotDeviceService iotDevice;

    public RobotController(Config config) {
        OpenHttpClient httpClient = new OpenHttpClient(config);
        this.iotDevice = new IotDeviceService(httpClient);
    }

    public IotDeviceService getIotDevice() {
        return iotDevice;
    }
}

